src/include/rados/librados.hpp

```diff
@@ -712,6 +712,14 @@ namespace librados
                    size_t len);
     int read(const std::string& oid, bufferlist& bl, size_t len, uint64_t off);
     int repair_read(const std::string& oid, bufferlist& bl, size_t len, uint64_t off, int flags, int32_t osdid, uint32_t e);
+    /**
+     * Rewrite the object with the replica hosted by specified osd
+     *
+     * @param osd from which OSD we will copy the data
+     * @param version the version of rewritten object
+     * @param what the flags indicating what we will copy
+     */
+    int repair_copy(const std::string& oid, uint64_t version, uint32_t what, int32_t osd, uint32_t epoch);
```

Comment:
This bit should probably be squashed into the later commit which changes it to take a set of bad shards, right?